includes/model/class-user.php

```diff
@@ -169,7 +169,14 @@ public function get_alternate_url() {
 	 * @return string The preferred username.
 	 */
 	public function get_preferred_username() {
-		return \get_the_author_meta( 'login', $this->_id );
+		$login = \get_the_author_meta( 'login', $this->_id );
+
+		// Handle cases where login is an email address (e.g., from Site Kit Google login).
+		if ( \filter_var( $login, FILTER_VALIDATE_EMAIL ) ) {
+			$login = \str_replace( array( '@', '.' ), '-', $login );
```

Comment:
this might be even better and should be the same result:

```suggestion
			$login = \get_the_author_meta( 'user_nicename', $this->_id );
```

Reply:
I guess it comes down to the expectations we have towards what makes for an appealing handle. a49da59021249d38848cd043302e50943fa763a3 moved away from `sanitize_title()` to transition fediverse handles from `usernamegmail-com` to `username-gmail-com`.

The latter might be a bit nicer, but happy to defer if you have another preference. I don't feel strongly about it.

Reply:
It might look nicer, but it breaks the lookup! If you stick with the custom formatter you would have to also add support to the Users::get_by_* functions

Reply:
Thanks that's helpful!